source/parser.hera

```diff
@@ -1037,15 +1038,17 @@ FunctionDeclaration
 
 FunctionSignature
   # NOTE: Merged in async and generator with optionals
-  ( Async __ )? Function Star? BindingIdentifier?:id __ NonEmptyParameters:parameters ReturnTypeSuffix?:suffix ->
+  ( Async __ )?:async Function:func ( __ Star )?:star ( __ BindingIdentifier )?:wid _?:w Parameters:parameters ReturnTypeSuffix?:suffix ->
```

Comment:
`BindingIdentifier` already includes `__` inside its productions.

Reply:
Ah, indeed. But we actually need a `_?` version, to support anonymous functions.

It'd be nice to have consistent names for rules that have or don't have whitespace at the start... The current grammar is a bit of a mix.  Maybe a leading `W` or something?

Reply:
Definitely worth exploring some conventions now that the parser is getting large.